### PR TITLE
TapArea: reimplement tapStyle

### PR DESCRIPTION
**Pull request by AlbertCarreras**
## Test Plan
- TapArea only compresses only with tapStyle='compress'
- Button compress behavior is not affected.

### CODE REVIEW

File: packages/gestalt/src/InternalLink.js [hunk 1]
```diff
@@ -81,6 +82,7 @@ const InternalLinkWithForwardRef: AbstractComponent<
     rel,
     rounding,
     size,
+    tapStyle = 'compress',
```
Comment (AlbertCarreras) [inline packages/gestalt/src/InternalLink.js:85]: default for Button/IconButton

File: packages/gestalt/src/TapArea.js [hunk 2]
```diff
@@ -89,6 +90,7 @@ const TapAreaWithForwardRef: React$AbstractComponent<
     onMouseLeave,
     onTap,
     rounding = 0,
+    tapStyle = 'none',
```
Comment (AlbertCarreras) [inline packages/gestalt/src/TapArea.js:93]: default is none for TapArea

File: docs/src/TapArea.doc.js [hunk 3]
```diff
@@ -220,26 +220,37 @@ card(
...
       },
+      {
+        name: 'tapStyle',
+        type: `none | 'compress'`,
+        required: false,
+        defaultValue: 'none',
+        description: [
+          'Set a compressing behavior when the TapArea is clicked / touched.',
+          `- 'none' does not compress TapArea.`,
+          `- 'compress' scales down TapArea by 1%.`,
```
Comment (christianvuerings) [inline docs/src/TapArea.doc.js:238]: @AlbertCarreras I would remove the `by 1%`

We actually compress it this way: https://github.com/pinterest/gestalt/blob/4f520db60d005670c8db7b6bee1500fa2e1a0419/packages/gestalt/src/useTapFeedback.js#L48

The reason for doing that is so we do not shrink large objects too much

Comment (AlbertCarreras) [reply docs/src/TapArea.doc.js:238]: DONE

File: docs/src/TapArea.doc.js [hunk 4]
```diff
@@ -220,26 +220,37 @@ card(
...
+        name: 'tapStyle',
+        type: `none | 'compress'`,
+        required: false,
+        defaultValue: 'none',
+        description: [
+          'Set a compressing behavior when the TapArea is clicked / touched.',
+          `- 'none' does not compress TapArea.`,
+          `- 'compress' scales down TapArea.`,
+        ],
+        href: 'tapStyle',
```
Comment (christianvuerings) [inline docs/src/TapArea.doc.js:240]: @AlbertCarreras this link doesn't seem to work - should it be `roles` instead?

Comment (AlbertCarreras) [reply docs/src/TapArea.doc.js]: DONE

**Review by christianvuerings — APPROVED** [review]

File: docs/src/TapArea.doc.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/src/TapArea.doc.js]: I was going to create it's own example, but add it there at the end.

**Pull request merged**